### PR TITLE
Basic footer for the Merlin's site

### PR DESCRIPTION
**Pull request by olibrook**
**JIRA**: 
- https://mobify.atlassian.net/browse/WEB-930
- https://mobify.atlassian.net/browse/WEB-922

## Changes
- A few bits of global setup (fonts, app-wide action for page loads).
- New footer.

## How to test-drive this PR
- Run locally and review against designs.

## Todos
- Pass on footer CSS/HTML

### CODE REVIEW

File: app/containers/footer/_footer.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+}
+
+.t-footer__panel {
+    padding: 15px;
+}
+
+.t-footer__panel--newsletter {
+    position: relative;
```
Comment (jeffkamo) [inline app/containers/footer/_footer.scss]: @olibrook what is this `position: relative` used for?

Comment (olibrook) [inline app/containers/footer/_footer.scss]: Fix display of the border

File: app/containers/footer/parsers/parser.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                title: 'Orders and Returns'
+            },
+            {
+                url: 'http://www.merlinspotions.com/catalogsearch/advanced/',
+                title: 'Advanced Search'
+            }
+        ]
+        expect(parser.parseNavigation($content)).toEqual(expected)
+    })
+})
```
Comment (DanielBrookRoberge) [inline app/containers/footer/parsers/parser.test.js]: I sure do love seeing these tests in the project!

File: app/utils/utils.js [hunk 3]
```diff
@@ -24,3 +24,17 @@ export const createAction = (description, ...argNames) => {
...
+/**
+ * Form-encode an arbitrary JS object.
+ */
+export const formEncode = (data) => {
+    const pairs = []
+    for (const k in data) {
+        if (data.hasOwnProperty(k)) {
+            pairs.push(`${encodeURIComponent(k)}=${encodeURIComponent(data[k])}`)
+        }
+    }
```
Comment (DanielBrookRoberge) [inline app/utils/utils.js]: We could replace this loop with `Object.keys(data).forEach()` which will only loop over the own properties. (ES5 so it needs no polyfill)

File: app/containers/footer/container.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            <div className="t-footer__social u-padding-md">
+                <div className="u-flexbox u-justify-center u-padding-md">
+                    {
+                        social.map((item) => {
+                            const [url, icon, title] = item
+                            return (
+                                <a href={url} className="t-footer__social-link" key={url}
+                                    style={{backgroundImage: `url(${getAssetUrl(icon)})`}}>
+                                    <span className="u-visually-hidden">{title}</span>
+                                </a>
```
Comment (DanielBrookRoberge) [inline app/containers/footer/container.js]: Are these icons not available through the Icon component?

Comment (jeffkamo) [inline app/containers/footer/container.js]: Unfortunately not at the moment. Shoppicon, the SDK's default icon set, does not include any social icons.

Whenever I get around to tackling the icons properly, I'll be sure to include the social icons too.

**Comment by jeffkamo** [comment]
I think I've applied everything I needed to add to this PR. 👍

**Comment by jeffkamo** [comment]
Except for perhaps that breaking test... looking into it.

File: app/containers/footer/actions.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+export const signUpToNewsletter = (action, method, data) => {
+    return (dispatch) => {
+
+        const onSuccess = () => {
+            dispatch(newsletterSignupComplete(constants.SIGNUP_SUCCESSFUL))
+        }
+
+        const onFail = () => {
+            dispatch(newsletterSignupComplete(constants.SIGNUP_FAILED))
+        }
```
Comment (olibrook) [inline app/containers/footer/actions.js]: I don't have grand plans, but we *are* supposed to do a notification thing on success, which is why I set it up the way it is. Gonna leave it, if that's cool.

Comment (DanielBrookRoberge) [inline app/containers/footer/actions.js]: Yeah that's absolutely fine, though I realize we usually structure this as 
```javascript
return utils.makeRequest(action, opts)
    .then(() => stuff)
    .catch(() => otherStuff)
```
which is why it looked odd to me.

File: app/utils/utils.js (hunk 3 above)
Comment (olibrook) [inline app/utils/utils.js]: Fixed

File: app/containers/footer/actions.js (hunk 5 above)
Comment (olibrook) [inline app/containers/footer/actions.js]: Oh right. I thought there was a preference generally for assigning names to these callbacks, for debugging purposes?

Comment (DanielBrookRoberge) [inline app/containers/footer/actions.js]: IIRC we still get a reasonable stack trace without that for something simple. But I could be wrong.

**Pull request merged**